R/Lrnr_base.R

```diff
@@ -19,28 +19,39 @@
 #'   }
 #' @importFrom assertthat assert_that is.count is.flag
 #' @importFrom uuid UUIDgenerate
+#' @importFrom memoise memoise cache_memory cache_filesystem
 
 #' @family Learners
 Lrnr_base <- R6Class(classname = "Lrnr_base",
                    portable = TRUE,
                    class = TRUE,
                    public = list(
-                     params = NULL,
-                     initialize = function(...) {
-                       params=list(...)
-                       if(length(params)==1 && names(params)=="params"){
-                         params=params$params
+                     initialize = function(params=NULL, memoise_learner = getOption("sl3.memoise.learner"), ...) {
+                       if(is.null(params)){
+                         params=list(...)
                        }
-
-                       self$params=params
+                       
+                       private$.params=params
                        private$.learner_uuid = UUIDgenerate(use.time=T)
                        
+                       if(memoise_learner){
```

Comment:
OK, I see, pretty cool. So if memoisation is turned on, the private$.train() method is essentially placed into the private$.memoised_train?

Reply:
I am still not following though in how does the `private$.learner_uuid` get used, in combination with `task$uuid`?

Reply:
yeah I moved the old `self$train` code into `self$default_train` and now `self$train` calls that or it's memoized version depending on if memoisation is turned on

Reply:
So memoisation happens within each learner instance. So the learner basically knows when it has been trained on the same or new data. That knowledge is based on uuid of the task? If I am getting a task with a uuid that has already been memoised, then I will just return the previous value. What happens if I get a task with new uuid? Does it get retrained and I am saving only the last fit object or am I saving all fit objects?

Reply:
private$.learner_uuid is actually not used at all here, you get a unique cache for each learner instance by calling memoise in the constructor for the learner. See the `memA2` example here: https://www.rdocumentation.org/packages/memoise/versions/1.1.0/topics/memoise